File: Testes a parte/Manipulação do Backend (Render)/por jS/acordarBanco.js
require('dotenv').config();
const { Client } = require('pg');

const MAX_TENTATIVAS = 5;
const INTERVALO = 5000; // 5 segundos

function getDbClient() {
  return new Client({
    host: process.env.DB_HOST,
    database: process.env.DB_NAME,
    user: process.env.DB_USER,
    password: process.env.DB_PASSWORD,
    port: process.env.DB_PORT,
    ssl: { rejectUnauthorized: false } // obrigatório para Render
  });
}

async function tentarConexao() {
  const client = getDbClient();
  try {
    await client.connect();
    const result = await client.query('SELECT 1;');
    console.log('✅ Banco acordado! Resultado:', result.rows);
    await client.end();
    return true;
  } catch (error) {
    console.log('⚠️  Erro ao tentar conexão:', error.message);
    return false;
  }
}

async function acordarBanco() {
  console.log('🔎 Verificando o status do banco PostgreSQL...');
  for (let tentativa = 1; tentativa <= MAX_TENTATIVAS; tentativa++) {
    console.log(`⏳ Tentativa ${tentativa}/${MAX_TENTATIVAS}...`);
    const sucesso = await tentarConexao();

    if (sucesso) {
      console.log('✅ Conexão estabelecida com sucesso!');
      return;
    }

    if (tentativa < MAX_TENTATIVAS) {
      console.log(`🕐 Aguardando ${INTERVALO / 1000} segundos para tentar novamente...`);
      await new Promise(resolve => setTimeout(resolve, INTERVALO));
    }
  }

  console.log('🚫 Não foi possível acordar o banco após várias tentativas.');
}

acordarBanco();

File: Testes a parte/Manipulação do Backend (Render)/por Python/acordarBanco.py
import psycopg2
from psycopg2 import OperationalError
from dotenv import load_dotenv
import os
import time

# Carrega variáveis do .env
load_dotenv()

DB_HOST = os.getenv("DB_HOST")
DB_NAME = os.getenv("DB_NAME")
DB_USER = os.getenv("DB_USER")
DB_PASSWORD = os.getenv("DB_PASSWORD")
DB_PORT = os.getenv("DB_PORT")


def conectar():
    return psycopg2.connect(
        host=DB_HOST,
        dbname=DB_NAME,
        user=DB_USER,
        password=DB_PASSWORD,
        port=DB_PORT,
        sslmode='require'  # importante para Render
    )


def verificar_banco():
    print("🔎 Verificando se o banco está acordado...")
    try:
        conexao = conectar()
        cursor = conexao.cursor()
        cursor.execute("SELECT 1;")
        resultado = cursor.fetchone()
        print("✅ Banco está acordado! Resultado:", resultado)
        cursor.close()
        conexao.close()
        return True
    except OperationalError as erro:
        print("⚠️ Banco parece estar dormindo ou inacessível.")
        return False


def acordar_banco(max_tentativas=5):
    tentativas = 0
    while tentativas < max_tentativas:
        tentativas += 1
        print(f"⏳ Tentando acordar o banco... (tentativa {tentativas})")
        try:
            conexao = conectar()
            cursor = conexao.cursor()
            cursor.execute("SELECT 1;")
            resultado = cursor.fetchone()
            print("✅ Banco acordado com sucesso. Resultado:", resultado)
            cursor.close()
            conexao.close()
            return True
        except OperationalError as erro:
            print("❌ Ainda não foi possível conectar. Esperando 5 segundos...")
            time.sleep(5)

    print("🚫 Não foi possível acordar o banco após várias tentativas.")
    return False


# Execução principal
if verificar_banco():
    print("🔁 Banco já está ativo, pronto para uso.")
else:
    acordar_banco()
